### PR TITLE
Add SubscriberClient.close() to examples

### PR DESCRIPTION
**Pull request by pradn**

### CODE REVIEW

**Comment by pradn** [comment]
The close() method was only recently added. Perhaps you're thinking of something else?

https://github.com/googleapis/python-pubsub/commit/b58d0d8e404c0a085b89d3407e6640651e81568c

**Comment by anguillanneuf** [comment]
@pradn Is it not this one here? 

https://googleapis.dev/python/pubsub/latest/subscriber/api/client.html#google.cloud.pubsub_v1.subscriber.client.Client.close

File: pubsub/cloud-client/quickstart/sub.py [hunk 1]
```diff
@@ -48,12 +48,14 @@ def callback(message):
...
     print("Listening for messages on {}..\n".format(subscription_path))
 
-    # Calling result() on StreamingPullFuture keeps the main thread from
-    # exiting while messages get processed in the callbacks.
-    try:
-        streaming_pull_future.result()
-    except:  # noqa
-        streaming_pull_future.cancel()
+    # Wrap subscriber in a 'with' block to automatically call close() when done.
+    with subscriber:
```
Comment (anguillanneuf) [inline pubsub/cloud-client/quickstart/sub.py]: Should this be `with client:`?

**Comment by anguillanneuf** [comment]
@pradn This branch has conflicts that must be resolved. I'm closing your PR and will work on a new one.

**Pull request closed without merging**